docs/source/quickstart.mdx

```diff
@@ -16,63 +16,125 @@ limitations under the License.
+from optimum.neuron import NeuronSFTConfig, NeuronSFTTrainer, NeuronTrainingArguments
+from optimum.neuron.models.training import NeuronModelForCausalLM
+
+
+def format_dolly_dataset(example):
+    """Format Dolly dataset into instruction-following format."""
+    instruction = f"### Instruction\n{example['instruction']}"
+    context = f"### Context\n{example['context']}" if example["context"] else None
+    response = f"### Answer\n{example['response']}"
+    
+    # Combine all parts with double newlines
+    parts = [instruction, context, response]
+    return "\n\n".join(part for part in parts if part)
+
+
+def main():
+    # Load instruction-following dataset
+    dataset = load_dataset("databricks/databricks-dolly-15k", split="train")
+    
+    # Model configuration
+    model_id = "Qwen/Qwen3-1.7B"
+    output_dir = "qwen3-1.7b-finetuned"
+    
+    # Setup tokenizer
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+    tokenizer.pad_token = tokenizer.eos_token
+    
+    # Configure training for Trainium
+    training_args = NeuronTrainingArguments(
+        learning_rate=1e-4,
+        tensor_parallel_size=8,  # Split model across 8 accelerators
+        per_device_train_batch_size=1,  # Batch size per device
+        gradient_accumulation_steps=8,
+        logging_steps=1,
+        output_dir=output_dir,
+    )
+    
+    # Load model optimized for Trainium
+    model = NeuronModelForCausalLM.from_pretrained(
+        model_id,
+        training_args.trn_config,
+        torch_dtype=torch.bfloat16,
+        use_flash_attention_2=True,  # Enable fast attention
+    )
+    
+    # Setup supervised fine-tuning
+    sft_config = NeuronSFTConfig(
+        max_seq_length=2048,
+        packing=True,  # Pack multiple samples for efficiency
+        **training_args.to_dict(),
+    )
+    
+    # Initialize trainer and start training
+    trainer = NeuronSFTTrainer(
+        model=model,
+        args=sft_config,
+        tokenizer=tokenizer,
+        train_dataset=dataset,
+        formatting_func=format_dolly_dataset,
+    )
+    
+    trainer.train()
+    
+    # Share your model with the community
+    trainer.push_to_hub(
+        commit_message="Fine-tuned on Databricks Dolly dataset",
+        blocking=True,
+        model_name=output_dir,
+    )
+    
+    if xr.local_ordinal() == 0:
+        print(f"Training complete! Model saved to {output_dir}")
+
+
+if __name__ == "__main__":
+    main()
 ```
 
-All Trainium instances come at least with 2 Neuron Cores. To leverage those we need to launch the training with `torchrun`.
-Below you see an example of how to launch a training script on a `trn1.2xlarge` instance using a `bert-base-uncased` model.
+This example demonstrates supervised fine-tuning on the [Databricks Dolly dataset](https://huggingface.co/datasets/databricks/databricks-dolly-15k) using `NeuronSFTTrainer` and `NeuronModelForCausalLM` - the Trainium-optimized versions of standard Transformers components.
```

Comment:
again `NeuronModelForCausalLM` of training but not inference, aie, we got to rework the issue fast.